mlir/lib/Dialect/Affine/Analysis/AffineStructures.cpp

```diff
@@ -222,8 +221,17 @@ LogicalResult FlatAffineValueConstraints::addBound(BoundType type, unsigned pos,
   fullyComposeAffineMapAndOperands(&map, &operands);
   map = simplifyAffineMap(map);
   canonicalizeMapAndOperands(&map, &operands);
-  for (auto operand : operands)
+  for (auto operand : operands) {
+    if (!((isTopLevelValue(operand)) || (isAffineInductionVar(operand)))) {
+      emitError(
+          operand.getLoc(),
+          " Above Val is neither defined as a Top-Level Value nor is an "
+          "AffineInductionVar. Non-terminal symbol / loop IV expected. Pass "
+          "Aborted!");
```

Comment:
Putting an emitError is not the right fix - this method is simply meant to return failure. It's the caller that has to deal with it appropriately.